### DISCUSSION TITLE
Training new entities to pre-trained "en_core_web_sm/lg" NER model, model not identifying entities that comes with default model.

### DISCUSSION THREAD

**Discussion by rammohanrrr** [explosion/spaCy#7395] [Help: Coding & Implementations]
nlp = spacy.load("en_core_web_sm")   
    ner = nlp.get_pipe("ner")
   
    for text, annotations in TRAIN_DATA:
        for ent in annotations.get("entities"):
            # print(text[ent[0]:ent[1]], ent[2])
            ner.add_label(ent[2])

    # get names of other pipes to disable them during training
    pipe_exceptions = ["ner", "trf_wordpiecer", "trf_tok2vec"]
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in pipe_exceptions]

    with nlp.disable_pipes(*other_pipes):  # only train NER
        examples = []
        optimizer = nlp.resume_training()
        for text, annots in TRAIN_DATA:
            examples.append(Example.from_dict(nlp.make_doc(text), annots))
        for itn in range(n_iter):
            random.shuffle(examples)
            losses = {}
            for batch in minibatch(examples, size=compounding(4.0, 32.0, 1.001)):
                nlp.update(batch, sgd=optimizer, losses=losses)
            print('losses : ', losses)

I want  a model which recognizes pre-defined entities that comes with model and new entities that I trained.
Please suggest me

Thanks

**Comment by svlandeg** [comment]
Hi! This is a good question for the discussions forum - so I'll move it there. You'll get a notification that this issue was closed, but it will forward automatically to the new thread on the forum that will remain open.

**Reply by rammohanrrr** [reply]
Is it a bug or program mistake,
Can you please give suggestions here

**Comment by svlandeg** [comment]
Hi!

Apologies for the late follow-up. 

To be honest, I'm not 100% sure I understand what you're asking - it would have been helpful to give a bit more context of what you're trying to do and what behaviour you're seeing, versus what behaviour you were expecting.

You can definitely add new entity types to an existing, pretrained model. However, if you would only feed in training examples of the new type, after some time the model will start "forgetting" what it learned before, because it's adjusting its internal weights to those new examples (only), and is not awarded/penalized for predictions of the old entity type(s). We call this "catastrophic forgetting" and I think if you search on that term here on the issue tracker and discussion board, you'll find more relevant information about that.

The typical advice is to make sure you mix in predictions of the old entity types as well - gold-standard annotations if you have them, or predictions generated with the original model, to prevent this.

I hope that answers your question - if not please let us know!
